### PR TITLE
Implements Departmental Security

### PR DESCRIPTION
**Pull request by LordFowl**
Removes the security officer role, and replaces it with five different roles, each with one slot each:
1 General Officer, who serves under the Warden and HoS, and can either man the brig or patrol.
1 Supply Officer, who serves under the Quartermaster and HoP, and is servant of cargo.
1 Science Officer, who serves under the RD and is servant of science
1 Engineering Officer, who serves under the CE and is servant of engineering'
1 Medical Officer, who serves under the CMO and is servant of medical.

These officers all have wide access to their own department, and access to the front doors of security. The HoS does not have direct control over any officer other than the General Officer, however heads of staff are more than encouraged to take his tactical advise as head of security affairs under keen consideration. Departmental officers only have access to their own departmental comms. Departmental cubicles have been mapped in, and the brig has been mapped down.

Currently the only way to discern a departmental officer is by their armband, their ID, and their colored helmet. I am looking into deeper uniform resprites.

- [x] Test access

- [ ] Fix power issues

- [ ] Consider access issues

image
image
image
image
image

https://forums.aurorastation.org/topic/11236-departmental-security/page/3/

### CODE REVIEW

**Comment by LordFowl** [comment]
!review

**Comment by LordFowl** [comment]
Cadets remain under the purview of HoS. It may or may not canonically make sense for the HoS to not have direct control but gameplay wise I believe it is necessary. In emergencies the HoS still has operational authority.

All security have access to the arrivals checkpoint.

**Comment by MarinaGryphon** [comment]
> Cadets remain under the purview of HoS. It may or may not canonically make sense for the HoS to not have direct control but gameplay wise I believe it is necessary. In emergencies the HoS still has operational authority.

Understandable, okay.

> All security have access to the arrivals checkpoint.

I don't know about this. To staff it would take away an officer from their own department.

**Comment by LordFowl** [comment]
I don’t see the checkpoint ever used except during departure

**Comment by MarinaGryphon** [comment]
> I don’t see the checkpoint ever used except during departure

Recently I've seen people manning it at the start of a round on code green, which is awful, yeah, but it still happens.

File: maps/aurora/aurora-4_mainlevel.dmm [hunk 1]
```diff
@@ -30040,13 +27669,21 @@
 	pixel_x = 0;
 	pixel_y = -32
 	},
+/obj/structure/cable/green{
+	icon_state = "1-2";
```
Comment (MarinaGryphon) [inline maps/aurora/aurora-4_mainlevel.dmm]:
```
## UNIT_TEST ##: --------------- floor [110 / 75 / 4] Contains multiple wires with same direction on top of each other.
## UNIT_TEST ##: !!! FAILURE !!! [MAP: Cable Test (Station)]: [1 / 7294] Some turfs had overlapping wires going the same direction.
```

Engineering office has a mapping issue.

**Comment by MarinaGryphon** [comment]
> You should still keep security vending machine inside of the brig. For things like handcuffs, grenades, etc. Departmental security offices also need a bit more extra decor like maybe a plant in the corner in each since they seems to have so much empty space. Also fire extinguisher cabinets seem to be missing in them.

They have potted plants now, as well as fire extinguishers, @Sindorman.

**Comment by XanderDox** [comment]
Why not just remove the Head of Security altogether since they're neutered, leave warrants to the Warden and emergency command to the Captain?

**Comment by MarinaGryphon** [comment]
> Why not just remove the Head of Security altogether since they're neutered, leave warrants to the Warden and emergency command to the Captain?

No.

**Comment by VTCobaltblood** [comment]
Medical office doesn't seem to have a holopad?

**Comment by OneOneThreeEight** [comment]
>Why not just remove the Head of Security altogether since they're neutered, leave warrants to the Warden and emergency command to the Captain?

No.

Onto the subject matter. In terms of design philosophy, the problem that really fractures most of the departmental security officers from the rest of the 'team' is that they do not have interconnected access to speak to one another over the security radio.

I would suggest, in order to address that fracturing issue, that each of the departmental officers receive Sec/PostedDepartment radios. The various department officers can facilitate communication, talk casually, and otherwise not be shafted in the respect of having so few options in a worst-case scenario.

**Comment by ben10083** [comment]
@Arrow768 can you add "feedback required" to this? This change is very controversial

**Comment by LordFowl** [comment]
This change already has a feedback thread. You've posted in it.

**Comment by ben10083** [comment]
@LordFowl yes, but the feedback provided in the thread should start to be more uniform, with a consensus of a affirmative to the Project, before a permanent merge. The tag only informs people about the PR in a feedback stage

**Comment by LordFowl** [comment]
wat

**Comment by MarinaGryphon** [comment]
> Onto the subject matter. In terms of design philosophy, the problem that really fractures most of the departmental security officers from the rest of the 'team' is that they do not have interconnected access to speak to one another over the security radio.
> 
> I would suggest, in order to address that fracturing issue, that each of the departmental officers receive Sec/PostedDepartment radios. The various department officers can facilitate communication, talk casually, and otherwise not be shafted in the respect of having so few options in a worst-case scenario. Not that my approval really matters - but I think a compromise is helpful, especially if it addresses the biggest weaknesses of something new being added.

I would advise to not make such suggestions based on possible issues before it has been tested and before people come up with in-character solutions to these issues. These would be best addressed by in-character policy, I think.

**Comment by LordFowl** [comment]
It is already screenshotted?

**Comment by dragoslava** [comment]
> It is already screenshotted?

nvm im blind. good boy fowl

**Comment by Mogelix** [comment]
1. As a departmental security officer, I don't get access to most if not all of the brig, which will make lone sec just this little bit worse. I also can't access security voidsuits in EVA. Oh look, I ALSO CANNOT ACCESS PROCESSING.
Seriously? Why would you deprive officers of the ability to process. Think a little. Not every rounds gonna have a fully stacked security team.
2. The brigs 'calm RP' places have been removed. The bathroom I quite frequently used to change my hairstyle, and I know many used it as a place to easily AFK. I also remember many fun RP interactions between officers occurred in the breakroom.
3. As a result of this PR which was supported for breaking departmental cliques and forcing security to RP more, security is now more isolated then ever. Any comradery they faced with other officers is no longer allowed as that would involve leaving their designated box or using the security radio they don't have. And you don't get much RP with your assigned department because they view you as a stupid brute/a intruder.
4. Departmental security doesn't know what it's doing. They can't call out "Intruder over here!" or they attract the entire station. They can't relay information to security as a whole without alerting everyone to the contents of what they're attempting to relay.
5. Departmental Security also doesn't know what's going on. If their department gets contact with intruders, they're department are the only ones that know, since alerting the rest of Depsec via common may be cumbersome or worsen the situation. Worse even, the few that do have access to security headsets may encounter the antags. So now it's not a situation of "Only security can RPwith the antags" its "Only command and security that has security channel can RP with antags."

Really just a bad idea. Drop it. Let it go. It might've been good in your head, but it's not worth enduring this crap.

**Comment by OneOneThreeEight** [comment]
This is not the feedback thread.

**Pull request closed without merging**